suzieq/poller/controller/source/netbox.py

```diff
@@ -160,16 +191,37 @@ async def get_inventory_list(self) -> List:
         Returns:
             List: inventory list
         """
-        url = f'{self._server.protocol}://{self._server.host}:'\
-            f'{self._server.port}/api/dcim/devices/?tag={self._data.tag}'
+
+        # if more than one tag is set, all devices that as at least one of the
+        # tags are selected. Netbox api doesn't allow to select with this
+        # logic by default. If the url query is constructed like
+        # '?tag=tag1&tag=tag2', only devices with both tags will be returned.
+        # For this reason, a different request for each tag must be performed.
+        # Devices with more than one tag may appear duplicated. It's also
+        # necessary to drop duplicates.
         if not self._session:
             headers = self._token_auth_header()
             self._init_session(headers)
+
+        # ids is used to avoid to return duplicated devices associated with
+        # multiple tags. This set contains the device ids of each device
+        # added to 'devices'
+        ids = set()
+        devices = []
         try:
-            devices, next_url = await self._get_devices(url)
-            while next_url:
-                cur_devices, next_url = await self._get_devices(next_url)
-                devices.extend(cur_devices)
+            for url in self._get_url_list():
+                logger.debug(f"Netbox: Retrieving url '{url}'")
+                url_devices, next_url = await self._get_devices(url)
+                while next_url:
+                    logger.debug(f"Netbox: Retrieving url '{next_url}'")
+                    cur_devices, next_url = await self._get_devices(next_url)
+                    url_devices.extend(cur_devices)
+                for device in url_devices:
+                    # remove duplicates
+                    dev_id = device.get('id')
+                    if dev_id is not None and dev_id not in ids:
+                        ids.add(dev_id)
+                        devices.append(device)
```

Comment:
We might simplify this piece of code just building a dictionary and returning the values of the dictionary:
```suggestion
               # Remove the duplicates from the list of received devices
              devices = list({device['id']: device for device in url_devices
                              if device.get('id') is not None}.values())
```

Reply:
This solution is not possible as is because devices must contain unique devices for each different tag. With the solution you proposed, `devices` will contain the result of only the last url. Anyway, I'll simplify the code with a solution similar to your suggestion. Thanks